README.rst

```diff
@@ -0,0 +1,419 @@
+
+While installing MPI and NCCL itself may seem like an extra hassle, it only needs to be done once by the team dealing
+with infrastructure, while everyone else in the company who builds the models can enjoy the simplicity of training them at
+scale.
+
+
+Install
+------------
+
+To install Horovod:
+
+1. Install `Open MPI <https://www.open-mpi.org/>`_ or another MPI implementation. Learn how to install Open MPI `on this page <https://www.open-mpi.org/faq/?category=building#easy-build>`_.
+
+**Note**: Open MPI 3.1.3 has an issue that may cause hangs.  The recommended fix is to
+downgrade to Open MPI 3.1.2 or upgrade to Open MPI 4.0.0.
+
+2. Install the ``horovod`` pip package.
+
+.. code-block:: bash
+
+    pip install horovod
+
+This basic installation is good for laptops and for getting to know Horovod.
+If you're installing Horovod on a server with GPUs, read the `Horovod on GPU <docs/gpus.md>`_ page.
+If you want to use Docker, read the `Horovod in Docker <docs/docker.md>`_ page.
+
+
+Concepts
+------------
+
+Horovod core principles are based on `MPI <http://mpi-forum.org/>`_ concepts such as *size*, *rank*,
+*local rank*, *allreduce*, *allgather* and, *broadcast*. See `this page <docs/concepts.md>`_ for more details.
+
+
+Usage
+-----
+
+To use Horovod, make the following additions to your program:
+
+1. Run ``hvd.init()``.
+
+2. Pin a server GPU to be used by this process using ``config.gpu_options.visible_device_list``.
+   With the typical setup of one GPU per process, this can be set to *local rank*. In that case, the first process on
+   the server will be allocated the first GPU, second process will be allocated the second GPU and so forth.
+
+3. Scale the learning rate by number of workers. Effective batch size in synchronous distributed training is scaled by
+   the number of workers. An increase in learning rate compensates for the increased batch size.
+
+4. Wrap optimizer in ``hvd.DistributedOptimizer``.  The distributed optimizer delegates gradient computation
+   to the original optimizer, averages gradients using *allreduce* or *allgather*, and then applies those averaged
+   gradients.
+
+5. Add ``hvd.BroadcastGlobalVariablesHook(0)`` to broadcast initial variable states from rank 0 to all other processes.
+   This is necessary to ensure consistent initialization of all workers when training is started with random weights or
+   restored from a checkpoint. Alternatively, if you're not using ``MonitoredTrainingSession``, you can simply execute
+   the ``hvd.broadcast_global_variables`` op after global variables have been initialized.
+
+6. Modify your code to save checkpoints only on worker 0 to prevent other workers from corrupting them.
+   This can be accomplished by passing ``checkpoint_dir=None`` to ``tf.train.MonitoredTrainingSession`` if
+   ``hvd.rank() != 0``.
+
+Example (see the `examples <examples/>`_ directory for full training examples):
+
+.. code-block:: bash
```

Comment:
Can you change all Python code blocks to be marked up as Python?

Reply:
I think `.. code-block:: python` is what you wanted. Let me know if I'm mistaken